sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/InstrumentationRegistryHelper.java

```diff
@@ -6,22 +6,46 @@
 package com.microsoft.appcenter.utils;
 
 import android.os.Bundle;
-import android.support.test.InstrumentationRegistry;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
 
 /**
  * Wraps InstrumentationRegistry class to enable mocking in unit tests.
  */
 public class InstrumentationRegistryHelper {
 
+    private static final String[] LOCATIONS = new String[]{"androidx.test.platform.app.InstrumentationRegistry",
+            "androidx.test.InstrumentationRegistry",
+            "android.support.test.InstrumentationRegistry"};
+
     /**
      * Get the instrumentation arguments from the InstrumentationRegistry. Wrapper exists for unit
      * tests.
      *
      * @return the instrumentation arguments.
-     * @throws LinkageError          if the class, method is not found or does not match, typically no test dependencies in release.
-     * @throws IllegalStateException if no argument Bundle has been registered.
+     * @throws IllegalStateException if it cannot call getArguments using any of the provided classes.
      */
-    public static Bundle getArguments() throws LinkageError, IllegalStateException {
-        return InstrumentationRegistry.getArguments();
+    public static Bundle getArguments() throws IllegalStateException {
+        Exception exception = null;
+
+        for (String location : LOCATIONS) {
+            try {
+                Class<?> aClass = getClass(location);
+                Method getArguments = aClass.getMethod("getArguments", (Class[]) null);
```

Comment:
Code analysis warning here which is actually true, this method does not have parameters.
Thus the code has to be:
```suggestion
                Method getArguments = aClass.getMethod("getArguments");
```

Was this code actually tested in a instrumentation environment with either AndroidX or support?

Reply:
The original code came from [our Espresso Extension](https://github.com/microsoft/AppCenter-Test-Espresso-Extensions/blob/7610ea64ffd4709beeebe038fed4a2bdb84057ab/src/main/java/com/microsoft/appcenter/espresso/Factory.java#L40) - I was confident it would work as I'm lead to believe that is working correctly. However the code has deviated quite a bit from my original push so I'll run some tests to make sure it works in practice.

Reply:
The signature is definitely incorrect for reflection, getArguments does not have a class array parameter.
You need to call getArguments without parameter.

Reply:
The `Class[]` has to be removed to resolve the warnings, you can skip varargs when calling a method without argument by reflection.